### DISCUSSION TITLE
why i see this block time for script if I use script tag with defer

### DISCUSSION THREAD

**Discussion by pogran** [GoogleChrome/lighthouse#15533] [Q&A]
For site https://xn--80aaalhzvfe9b4a.xn--p1ai/ I see block time for script yandex.metrika with props **defer** how it possible?

![image](https://github.com/GoogleChrome/lighthouse/assets/5382042/77b1a3c6-511c-410c-bd28-b4b26fb77249)
![image](https://github.com/GoogleChrome/lighthouse/assets/5382042/b9a5f15c-fec3-4b4e-878a-ffec1293fe8e)

**Comment by adamraine** [comment]
The `defer` attribute doesn't prevent the script from being run. It just defers it's execution until after the page is finished parsing:
https://www.w3schools.com/tags/att_script_defer.asp

Lighthouse still analyzes the page after the page is parsed, giving the script plenty of time to run and block the main thread.

**Reply by pogran** [reply]
@adamraine  ok for my case what need to do with this? how to add metrics in code without block main thread?

**Reply by adamraine** [reply]
Ultimately, the script execution is what is blocking the main thread. Please see our existing docs on this topic:
https://developer.chrome.com/docs/lighthouse/performance/bootup-time/#how-to-speed-up-javascript-execution

**Reply by pogran** [reply]
@adamraine i see this point, but it not my script) i need to set setTimeout (i think it not a good chooise) for this if i want not to block main thread or mb i can use some events which tell me that main tread is free?

**Reply by adamraine** [reply]
Perhaps this will help: https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback

**Reply by pogran** [reply]
@adamraine no the same situation
![image](https://github.com/GoogleChrome/lighthouse/assets/5382042/065e793f-3b01-45e3-bf19-359ca63f0947)
![image](https://github.com/GoogleChrome/lighthouse/assets/5382042/15f6cb74-957d-4e49-9ed1-34fe78cc1af7)
